packages/crawler/src/crawler/puppeteer-crawler-engine.ts

```diff
@@ -60,6 +60,11 @@ export class PuppeteerCrawlerEngine {
             this.crawlerConfiguration.setChromePath(crawlerRunOptions.chromePath);
         }
 
+        if (crawlerRunOptions.singleWorker === true) {
```

Comment:
Do we have unit test code that validates this behavior? The current test code replicates this behavior but that's not the same as validating it.

Reply:
yes we do, in the PR I added new test cases to validate this code